ta/pkcs11/src/pkcs11_token.c

```diff
@@ -230,3 +230,138 @@ uint32_t entry_ck_token_info(uint32_t ptypes, TEE_Param *params)
 
 	return PKCS11_CKR_OK;
 }
+
+#if TRACE_LEVEL >= TRACE_DEBUG
+static void dmsg_print_supported_mechanism(unsigned int token_id,
+					   uint32_t *mecha_array, size_t count)
+{
+	size_t n = 0;
+
+	for (n = 0; n < count; n++)
+		DMSG("PKCS11 token %"PRIu32": mechanism 0x%04"PRIx32": %s",
+		     token_id, mecha_array[n], id2str_proc(mecha_array[n]));
+}
+#else
+static void dmsg_print_supported_mechanism(unsigned int token_id __unused,
+					   uint32_t *mecha_array __unused,
+					   size_t count __unused)
+{
+}
+#endif
```

Comment:
Can we do this without the `#if` clauses? I am fairly confident that the compiler will emit the same code for
```
static void dmsg_print_supported_mechanism(unsigned int token_id,
					   uint32_t *mecha_array, size_t count)
{
	size_t n = 0;

	if (TRACE_LEVEL < TRACE_DEBUG)
		return;

	for (n = 0; n < count; n++)
		DMSG("PKCS11 token %"PRIu32": mechanism 0x%04"PRIx32": %s",
		     token_id, mecha_array[n], id2str_proc(mecha_array[n]));
}
```

Reply:
ok, fine with me. thanks.